File: Hackerrank/python/BasicPrograms/max_consecutive_ones.py
def count_max_consecutive_ones(str):
    count = 0
    res = 0
    for i in range(0, len(str)):
        if str[i] == 0:
            count = 0
        else:
            count = count+1
            res = max(res, count)
    return res

str = [1,1,1,0,0,0,0,0,1,1,1,1,1]
print count_max_consecutive_ones(str)

File: LeetCode/EasyJava/MaxConsecutiveOnes.java
package Easy;

public class MaxConsecutiveOnes {

    public int maxConsecutiveOnes(int[] arr){
        int res = 0;
        int count  = 0;
        for(int i = 1; i < arr.length ; i++){
            if (arr[i] == 0)
                count = 0;
            else{
                if (arr[i-1] == arr[i]){
                    count++;
                    res = Math.max(res, count);
                }

            }
        }
        return res;
    }

}
